In [ ]:
#DATE
#USERS
#WIKI PAGES
#RESEARCH NOTES
#COMMENTS
#QUESTION
#ANSWERS

In [176]:
def path_to_url(nid, data_type, paths):
    
    try:

        path_type = (paths['type'] == data_type)
        path_nid = (paths['nid'] == nid)

        path = paths.loc[path_type & path_nid]['path'].values[0]

        return 'https://publiclab.org' + path
    
    except:
        
        return False

In [1]:
def create_csv_paths(data_types, base='', exp='', full=True):
    
    csv_paths = {}
    
    for data_type in data_types:

        if not full:
            csv_dir = '../data/2016_2019/samples/'
            e = '_' + str(base) + 'e' + str(exp)
            csv_path = csv_dir + data_type + e + '.csv'
        else:
            csv_dir = '../data/2016_2019/full/'
            e = ''
            csv_path = csv_dir + data_type + e + '.csv'
            
        csv_paths[data_type] = csv_path
    
    return csv_paths

In [180]:
import pandas as pd
import numpy as np
import re
import csv

sample = False

m = 25
exp = 2



if sample:
    csv_dir = '../data/2016_2019/samples/'
    e = '_' + str(m) + 'e' + str(exp)
    csv_path = csv_dir + data_type + e + '.csv'
else:
    csv_dir = '../data/2016_2019/full/'
    e = ''
    csv_path = csv_dir + data_type + e + '.csv'

data_types = ['answers', 'comments', 'notes', 'questions', 'users', 'wikis', 'paths', 'slugs']
csv_paths = create_csv_paths(data_types)

# print csv_paths

### LOAD PATHS

In [181]:
paths = pd.read_csv(csv_paths['paths'])
path_to_url(13745, 'notes', paths)

'https://publiclab.org/notes/Mushroomman/12-05-2016/is-anyone-doing-any-work-with-fungi-or-bioremediation'

### WIKI PAGES

In [84]:
wiki_cols = [u'nid',  u'uid', u'created', u'comments_count', u'cached_likes', u'title']
renamed_wiki_cols = {'created': 'date', 'comments_count': 'comments', 'cached_likes': 'likes'}

wikis = pd.read_csv(csv_paths['wikis'], usecols=wiki_cols)
wikis = wikis[wiki_cols] # questions.columns = cols
wikis.rename(index=str, columns=renamed_wiki_cols, inplace=True)

# wikis.columns
wikis.head()

nid     uid        date  comments  likes  \
0  10388       7  1525745225         0     27   
1  13290       7  1468599610         0      1   
2  13291  461120  1468610199         0      1   
3  13304  461120  1469191106         0      1   
4  13331  422561  1470259324         0      6   

                                             title  
0                                         OpenHour  
1                                      Riffle pics  
2  How To Obtain Transmittance/ Absorbance Spectra  
3                          Solux 4700K Lamp Set Up  
4                                   Public Comment

### RESEARCH NOTES

In [189]:
notes_cols = [u'nid',  u'uid', u'created', u'comments_count', u'cached_likes', u'title']
renamed_notes_cols = {'created': 'date', 'comments_count': 'comments', 'cached_likes': 'likes'}

notes = pd.read_csv(csv_paths['notes'])#, usecols=notes_cols)

notes = notes[notes_cols]
notes.rename(index=str, columns=renamed_notes_cols, inplace=True)

# notes.columns
notes.head()

nid     uid        date  comments  likes  \
0  13252  481667  1467398556         2      0   
1  13253  447491  1467400850         5      1   
2  13254  447491  1467415364         7      3   
3  13255       4  1467415531         2      1   
4  13256       4  1467418233         3      0   

                                               title  
0  Desktop spectrometry 3.0 kit: Can it quantify/...  
1       Capturing location information with trap cam  
2                             Mobius TrapCam Housing  
3          ideal mechanisms & silly putty camera rig  
4  Question: Does the Levitation Light dive for y...

### COMMENTS

In [88]:
cmnt_cols = ['nid', 'uid', 'cid', 'aid', 'timestamp', 'thread', 'comment', 'reply_to']

comments = pd.read_csv(csv_paths['comments'], usecols=cmnt_cols)
comments['reply_to'].fillna(0, inplace=True)
comments['reply_to'] = comments['reply_to'].astype('int64')
comments = comments[cmnt_cols]
comments.rename(index=str, columns={'timestamp': 'date'}, inplace=True)

# comments.columns
comments.head()

nid     uid    cid  aid        date thread  \
0  12876  447491  14837    0  1467331379    02/   
1  13248  461120  14838    0  1467363313    02/   
2  13248  461120  14839    0  1467382535    02/   
3  13248  461120  14840    0  1467382846    02/   
4  13248   54025  14841    0  1467396699    02/   

                                             comment  reply_to  
0  Just a catchup here:\r\n\r\nI've continued to ...         0  
1  @stoft hey Dave, I thought we never say "oops,...         0  
2  @stoft Hey Dave I thought you might find this ...         0  
3  @stoft hey dave I'm including the 2 plots agai...         0  
4  Unfortunately ..... while photomultipliers res...         0

### QUESTIONS

In [222]:
quest_cols = [u'nid', u'uid', u'qid', u'created', u'comment', u'comments_count', u'cached_likes', u'title']

renamed_quest_cols = {'created': 'date', 'comments_count': 'thread_cmnts', 
                      'comment': 'user_cmnts', 'cached_likes': 'likes'}

questions = pd.read_csv(csv_paths['questions'])#, usecols=quest_cols)
questions['qid'] = range(1, questions.shape[0] + 1)

questions = questions[quest_cols] # questions.columns = cols
questions.rename(index=str, columns=renamed_quest_cols, inplace=True)

# questions.columns
questions.head()

nid     uid  qid        date  user_cmnts  thread_cmnts  likes  \
0  13252  481667    1  1467398556           2             2      0   
1  13256       4    2  1467418233           2             3      0   
2  13286  483452    3  1468382733           2             7      0   
3  13312  483438    4  1469617059           2             3      1   
4  13358  460486    5  1471415749           2            12      0   

                                               title  
0  Desktop spectrometry 3.0 kit: Can it quantify/...  
1  Question: Does the Levitation Light dive for y...  
2                          Spectrometer from a drone  
3                                  Question: Joining  
4  How to neutralize/remove the source's spectra ...

### ANSWERS

In [183]:
ans_cols = [u'nid', u'uid', u'id', u'created_at', u'cached_likes', u'content', u'accepted']

answers = pd.read_csv(csv_paths['answers'])
answers = answers[ans_cols]

renamed_ans_cols = {'id': 'aid', 'created': 'date', 'cached_likes': 'likes'}
answers.rename(index=str, columns=renamed_ans_cols, inplace=True)

# answers.columns
answers.head()

nid     uid  aid               created_at  likes  \
0  13312  468506    2  2016-08-02 09:16:09 UTC      1   
1  13356       1    3  2016-08-16 23:04:23 UTC      0   
2  13356       7    4  2016-08-17 12:35:13 UTC      0   
3  13358       1    5  2016-08-17 14:27:28 UTC      0   
4  13358     554    6  2016-08-17 14:50:31 UTC      0   

                                             content  accepted  
0  Sorry to bother you. I was just testing the ne...      True  
1  Looks like it does indeed work... now to see i...      True  
2                      Here's an answer for you: 42!     False  
3  Hi, CCDs do not have a linear response curve -...      True  
4  I agree with Jeff that those results seem to s...      True

In [305]:
questions.shape

(820, 8)

### DATA RELATIONSIHPS

#### QUESTIONS VS. NOTES

In [288]:
questions.loc[questions['nid'] == 13745]
# print questions.loc[questions['nid'] == 18446]

nid     uid  qid        date  user_cmnts  thread_cmnts  likes  \
80  13745  498969   81  1480917522           2             3      0   

                                                title  
80  Is anyone doing any work with fungi? or biorem...

A QUESTION IS A NOTE WITH A POWER TAG OF QUESTION

In [289]:
notes.loc[notes['nid'] == 13745]
# print notes.loc[notes['nid'] == 18446]

nid     uid        date  comments  likes  \
349  13745  498969  1480917522         3      0   

                                                 title  
349  Is anyone doing any work with fungi? or biorem...

#### ANSWERS VS. COMMENTS

In [290]:
answers.loc[answers['nid'] == 13745]
# print answers.loc[answers['nid'] == 18446]

nid     uid  aid               created_at  likes  \
137  13745  499993  149  2016-12-15 00:59:26 UTC      1   
236  13745  237313  251  2017-06-06 19:08:04 UTC      0   

                                               content  accepted  
137  Hey Mushroomman! \r\n\r\nFellow mycology fanat...     False  
236  I'm late to the party, but a very cool topic! ...     False

ALL DISCUSSION ABOUT A QUESTION, INCLUDING ANSWERS AND REPLIES IN ALL LEVELS OF HIERARCHIES ARE COMMENTS.

HOWEVER, AN ANSWER IS *TECHNICALLY* A COMMENT OF THE HIGHEST LEVEL THREAD.

**NOTICE:** *AID 149 AND 251 ABOVE* ARE THE SAME AS *CID 22382 and 22481 BELOW*.  HOWEVER, *CID 22223 BELOW* IS IN REPLY TO *AID 149 (ABOVE AND BELOW) AND CID 22382 (BELOW)* WHICH ARE ALL THE SAME.  

In [249]:
comments.loc[comments['nid'] == 13745]
# comments.loc[comments['nid'] == 18446]

nid     uid    cid  aid        date thread  \
6374  13745  579767  22223  149  1552994882    NaN   
6526  13745  499993  22382    0  1481763566    /01   
6625  13745  237313  22481    0  1496776084    /01   

                                                comment  reply_to  
6374   Hello Jlmaybach.\r\n\r\nI am actually doing a...     22382  
6526  Hey Mushroomman! \r\n\r\nFellow mycology fanat...         0  
6625  I'm late to the party, but a very cool topic! ...         0

#### WIKI PAGES

WIKI PAGES DO NOT APPEAR TO HAVE ANY SUBTYPES SUCH AS WITH QUESTIONS & NOTES AND ANSWERS & COMMENTS

In [112]:
wikis.loc[wikis['nid'] == 13745]

Empty DataFrame
Columns: [nid, uid, date, comments, likes, title]
Index: []

In [ ]:
DATES -> WIKI_DATES + NOTES_DATES + COMMENT_DATES
USERS -> USERS_COUNT
WIKI PAGES -> WIKIS
RESEARCH NOTES -> NOTES - QUESTIONS
COMMENTS -> WIKI_COMMENTS + NOTE_COMMENTS + ANSWERS
QUESTIONS -> NOTES WHERE QUESTION_NID EQUALS NOTES_NID
ANSWERS -> COMMENTS WHERE QUESTIONS_NID EQUALS COMMENTS_NID

USER_WIKI_PAGES -> USERS_UID == WIKI_PAGES_UID
USER_NOTES -> USERS_UID == NOTES_UID
USER_COMMENTS -> USERS_UID == WIKI_PAGES_UID
USER_QUESTIONS -> USERS_UID == QUESTIONS_UID
USER_ANSWERS -> USERS_UID == ANSWERS_UID

WIKI PAGE COMMENTS -> COMMENTS WHERE WIKI_NID EQUALS COMMENTS_NID
RESEARCH NOTE COMMENTS -> COMMENTS WHERE NOTE_NID EQUALS COMMENTS_NID

### MATRIX NETWORKS

#### QUESTIONS AND ANSWERS

##### Q&A TEST

In [228]:
questions.loc[questions['nid'] == 13745]#['nid']#['uid']['qid'] 

nid     uid  qid        date  user_cmnts  thread_cmnts  likes  \
80  13745  498969   81  1480917522           2             3      0   

                                                title  
80  Is anyone doing any work with fungi? or biorem...

In [225]:
answers.loc[answers['nid'] == 13745]

nid     uid  aid               created_at  likes  \
137  13745  499993  149  2016-12-15 00:59:26 UTC      1   
236  13745  237313  251  2017-06-06 19:08:04 UTC      0   

                                               content  accepted  
137  Hey Mushroomman! \r\n\r\nFellow mycology fanat...     False  
236  I'm late to the party, but a very cool topic! ...     False

In [230]:
q = questions.loc[questions['nid'] == 13745]
a = answers.loc[answers['nid'] == 13745]

q_cols = ['nid', 'uid', 'qid']
a_cols = ['nid', 'uid', 'aid']

q_and_a = pd.merge(q[q_cols], a[a_cols], on='nid', how='outer')
q_and_a.columns = ['nid', 'quid', 'qid', 'auid', 'aid']
q_and_a

nid    quid  qid    auid  aid
0  13745  498969   81  499993  149
1  13745  498969   81  237313  251

In [231]:
questions_answers = pd.merge(questions[q_cols], answers[a_cols], on='nid', how='outer')
questions_answers.columns = ['nid', 'quid', 'qid', 'auid', 'aid']
questions_answers.head()

nid      quid  qid      auid  aid
0  13252  481667.0  1.0       NaN  NaN
1  13256       4.0  2.0       NaN  NaN
2  13286  483452.0  3.0       NaN  NaN
3  13312  483438.0  4.0  468506.0  2.0
4  13358  460486.0  5.0       1.0  5.0

In [232]:
path_to_url(13252, 'notes', paths)

'https://publiclab.org/notes/peace/07-01-2016/desktop-spectrometry-3-0-kit-can-it-quantify-identify-a-mixture-of-chemical-solvents'

In [233]:
notes[notes['nid'] == 13252]

nid     uid        date  comments  likes  \
0  13252  481667  1467398556         2      0   

                                               title  
0  Desktop spectrometry 3.0 kit: Can it quantify/...

In [234]:
comments[comments['nid'] == 13252]

nid     uid    cid  aid        date thread  \
14   13252       1  14851    0  1467726894    01/   
188  13252  481667  15034    0  1469564712    02/   

                                               comment  reply_to  
14   Hi, I haven't heard of someone doing this sort...         0  
188  Thank you for your comment.  That's a good poi...         0

In [250]:
comments[(comments['aid'] == 0) & (comments['thread'] == '01/')].shape

(1212, 8)

In [251]:
comments[(comments['aid'] == 0) & (comments['reply_to'] == 0)].shape

(6866, 8)

In [238]:
comments.shape

(7771, 8)

In [236]:
answers[answers['nid'] == 13252]#.shape

Empty DataFrame
Columns: [nid, uid, aid, created_at, likes, content, accepted]
Index: []

In [239]:
questions.shape

(820, 8)

In [311]:
qnids = list(questions['nid'])
q_resp = comments.loc[(comments['nid'].isin(qnids))]
q_resp_top_thread = comments.loc[(comments['nid'].isin(qnids)) & (comments['reply_to'] == 0)]
q_resp_sub_threads = comments.loc[(comments['nid'].isin(qnids)) & (comments['reply_to'] != 0)]
print 'TOTAL COMMENTS:', comments.shape[0]
print 'TOTAL COMMENTS RELATED TO A QUESTION:', q_resp.shape[0]
print 'TOTAL COMMENTS RELATED TO A QUESTION AND IN TOP THREAD', q_resp_top_thread.shape[0]
print 'TOTAL COMMENTS RELATED TO A QUESTION AND IN A SUB THREAD', q_resp_sub_threads.shape[0]

TOTAL COMMENTS: 7771
TOTAL COMMENTS RELATED TO A QUESTION: 2950
TOTAL COMMENTS RELATED TO A QUESTION AND IN TOP THREAD 2138
TOTAL COMMENTS RELATED TO A QUESTION AND IN A SUB THREAD 812


In [361]:
anids = list(answers['nid'])
c_not_in_a = q_resp_top_thread[~(q_resp_top_thread['nid'].isin(anids))]
reply_not_in_a = q_resp_sub_threads[~(q_resp_sub_threads['nid'].isin(anids))]
len(c_not_in_a['nid'].unique())#[:5]#.head()#shape[0] #502
reply_not_in_a['nid'].unique().shape

(7,)

In [359]:
c_not_in_a.head()

nid     uid    cid  aid        date thread  \
8   13256  421042  14845    0  1467421274    01/   
9   13256       4  14846    0  1467490094    02/   
10  13256  421042  14847    0  1467530783    02/   
14  13252       1  14851    0  1467726894    01/   
83  13286     554  14927    0  1468423282    01/   

                                              comment  reply_to  
8   <iframe width="560" height="315" src="https://...         0  
9   that is a very heavy tail.  I got a similar re...         0  
10  I doubt it would have much effect on a large d...         0  
14  Hi, I haven't heard of someone doing this sort...         0  
83  Hi Sylvain,\r\n\r\nI don't know the answer to ...         0

In [349]:
qnids = list(questions['nid'])
tlnids = list(q_resp_top_thread['nid'].unique())

len(tlnids)
# q_not_in_a = answers[~(answers['nid'].isin(qnids))]
# q_not_in_a.shape[0]

c_not_in_a = answers[~(answers['nid'].isin(tlnids))]
c_not_in_a.shape[0]

# tlnids_not_in_ans = answers[~(answers['nid'].isin(tlnids))]

# tlnids_not_in_ans.shape[0]
# q_not_in_a.shape[0]

q_not_in_a#['nid'].unique()
# c_not_in_a

nid     uid   aid               created_at  likes  \
1    13356       1     3  2016-08-16 23:04:23 UTC      0   
2    13356       7     4  2016-08-17 12:35:13 UTC      0   
13   13362  436738    21  2016-08-18 15:14:40 UTC      1   
14   13362  447988    22  2016-08-18 20:49:47 UTC      0   
15   13362     379    23  2016-08-21 23:44:22 UTC      0   
16   13362     379    24  2016-08-21 23:46:04 UTC      0   
17   13362     379    25  2016-08-21 23:47:50 UTC      0   
18   13362     379    26  2016-08-21 23:50:39 UTC      0   
20   13234       1    29  2016-08-31 14:25:14 UTC      0   
21   13115       1    30  2016-08-31 14:28:37 UTC      0   
27   13420  468506    36  2016-09-06 09:53:50 UTC      1   
28   13421  468506    37  2016-09-06 10:06:39 UTC      1   
34   12978  479194    43  2016-09-14 23:01:07 UTC      0   
35   13061  479194    44  2016-09-14 23:03:22 UTC      0   
208  14068  423961   221  2017-03-30 06:38:09 UTC      1   
306  11301  512046   326  2017-08-03 15:41:10 UTC      0   
307  10574  512046   327  2017-08-03 15:43:48 UTC      0   
324  10993  513569   353  2017-08-31 07:43:00 UTC      0   
325  10996  513569   354  2017-08-31 07:47:27 UTC      0   
327  11270  513632   356  2017-09-01 11:59:35 UTC      0   
328  11226  513632   357  2017-09-01 12:02:42 UTC      0   
329  11301  513632   358  2017-09-01 12:06:51 UTC      0   
330  10537  513632   359  2017-09-01 12:29:02 UTC      0   
466  15152  515171   539  2017-11-07 00:11:42 UTC      0   
468  15152       7   541  2017-11-09 20:51:09 UTC      0   
606  15538  237313   711  2018-01-17 17:18:17 UTC      0   
609  12436   54025   714  2018-01-17 22:24:31 UTC      1   
610  12436     554   715  2018-01-17 22:24:44 UTC      1   
611  11135     554   716  2018-01-17 22:25:08 UTC      0   
613  10984     554   718  2018-01-17 22:37:22 UTC      0   
669  10996  524670   779  2018-02-08 11:33:30 UTC      0   
684  11516       1   798  2018-02-16 21:45:38 UTC      0   
706  15809       1   821  2018-02-23 14:00:27 UTC      0   
727  15864       7   847  2018-03-05 19:03:26 UTC      1   
943      0  515081  1108  2018-08-03 20:14:34 UTC      0   
944      0  515081  1109  2018-08-03 20:16:34 UTC      0   

                                               content  accepted  
1    Looks like it does indeed work... now to see i...      True  
2                        Here's an answer for you: 42!     False  
13   The science is in on this question. The Earth ...      True  
14   Maybe Public Lab can help the public grow in u...     False  
15   I second the AR5 report, and especially the [S...     False  
16   I second the AR5 report, and especially the [S...     False  
17   Other scientific sites are [NASA ](http://clim...     False  
18   For the denial bit, I follow the [Yale climate...     False  
20   Yes, the live capture has been limited to 640p...     False  
21   (Moving my comment to an "answer" -- would you...      True  
27                               This is a demo answer      True  
28                               This is a demo answer      True  
34   @warren is there a chance to write a note abou...     False  
35   @Georgi did you ever end up changing the camer...     False  
208  @alanrday I don't know if this is helpful, but...     False  
306  Thank you guys - was wondering the same too[.]...     False  
307  Thanks for the info[.](http://www.dansk-firmai...     False  
324  Hope is the best thing when you see that there...     False  
325  It is really very [awesome](http://allbankrout...     False  
327  We do not disclose your personal information t...     False  
328  If you were to bring your own IR source (most ...     False  
329  Authorized disclosures include limited circums...     False  
330  We do not disclose your personal information t...     False  
466  [![P1130837.jpg](https://publiclab.org/system/...     False  
468  [![IMG_20171104_114716427.jpg](https://publicl...     False  
606  Not sure how micro you want to go, but 

In [344]:
questions[questions['nid'] == 13234]

Empty DataFrame
Columns: [nid, uid, qid, date, user_cmnts, thread_cmnts, likes, title]
Index: []

In [336]:
answers[answers['nid'] == 13356]

nid  uid  aid               created_at  likes  \
1  13356    1    3  2016-08-16 23:04:23 UTC      0   
2  13356    7    4  2016-08-17 12:35:13 UTC      0   

                                             content  accepted  
1  Looks like it does indeed work... now to see i...      True  
2                      Here's an answer for you: 42!     False

In [337]:
comments[comments['nid'] == 13356]

nid  uid    cid  aid        date thread  \
290   13356    7  15143    3  1471441378    NaN   
6390  13356    1  22246    0  1471388663    /01   
6391  13356    7  22247    0  1471437313    /01   

                                                comment  reply_to  
290            i'm commenting on this accepted answer.      22246  
6390  Looks like it does indeed work... now to see i...         0  
6391                      Here's an answer for you: 42!         0

In [323]:
anids = list(answers['nid'])
c_in_a = q_resp_top_thread[(q_resp_top_thread['nid'].isin(anids))]
c_not_in_a = q_resp_top_thread[~(q_resp_top_thread['nid'].isin(anids))]

c_in_a.shape[0], c_not_in_a.shape[0]
#(1636, 502) 1134


# q_resp = comments.loc[(comments['nid'].isin(qnids))]
# q_resp_sub_threads = comments.loc[(comments['nid'].isin(qnids)) & (comments['reply_to'] != 0)]

(1636, 502)

In [324]:
questions.loc[questions['nid'] == 13358]

nid     uid  qid        date  user_cmnts  thread_cmnts  likes  \
4  13358  460486    5  1471415749           2            12      0   

                                               title  
4  How to neutralize/remove the source's spectra ...

In [325]:
c_in_a[c_in_a['nid'] == 13358]#.head()

nid     uid    cid  aid        date thread  \
291   13358     554  15144    0  1471445049    01/   
292   13358     554  15145    0  1471445068    02/   
293   13358     554  15146    0  1471445467    02/   
294   13358     554  15147    0  1471445500    02/   
296   13358     554  15149    0  1471445702    02/   
6392  13358       1  22248    0  1471444048    /01   
6393  13358     554  22249    0  1471445431    /01   
6394  13358     554  22250    0  1471449778    /01   
6395  13358     554  22251    0  1471449853    /01   
6397  13358  460486  22253    0  1471461631    /01   
6398  13358  461120  22254    0  1471469589    /01   
6399  13358  461120  22255    0  1471470172    /01   

                                                comment  reply_to  
291   I agree with Jeff that those results seem to s...         0  
292   I agree with Jeff that those results seem to s...         0  
293                         Does this comment box work?         0  
294                                        Yes it does.         0  
296   Sorry, I am getting strange behavior from the ...         0  
6392  Hi, CCDs do not have a linear response curve -...         0  
6393  I agree with Jeff that those results seem to s...         0  
6394  [![ccdtrace.JPG](//i.publiclab.org/system/imag...         0  
6395  [![ccdDiff.JPG](//i.publiclab.org/system/image...         0  
6397  Do you think we need to fix the offset of magn...         0  
6398  Hey @shubham, first lets start with what your ...         0  
6399  Hey @shubham, this may help you out also:\r\nN...         0

In [320]:
answers.loc[answers['nid'] == 13358]

nid     uid  aid               created_at  likes  \
3   13358       1    5  2016-08-17 14:27:28 UTC      0   
4   13358     554    6  2016-08-17 14:50:31 UTC      0   
5   13358     554    7  2016-08-17 16:02:58 UTC      0   
6   13358     554    8  2016-08-17 16:04:13 UTC      0   
8   13358  460486   12  2016-08-17 19:20:31 UTC      0   
9   13358  461120   13  2016-08-17 21:33:09 UTC      0   
10  13358  461120   14  2016-08-17 21:42:52 UTC      0   

                                              content  accepted  
3   Hi, CCDs do not have a linear response curve -...      True  
4   I agree with Jeff that those results seem to s...      True  
5   [![ccdtrace.JPG](//i.publiclab.org/system/imag...      True  
6   [![ccdDiff.JPG](//i.publiclab.org/system/image...      True  
8   Do you think we need to fix the offset of magn...      True  
9   Hey @shubham, first lets start with what your ...     False  
10  Hey @shubham, this may help you out also:\r\nN...     False

In [326]:
c_not_in_a[c_not_in_a['nid'] == 13358]

Empty DataFrame
Columns: [nid, uid, cid, aid, date, thread, comment, reply_to]
Index: []

In [267]:
q_resp_top_thread.head()

nid     uid    cid  aid        date thread  \
8   13256  421042  14845    0  1467421274    01/   
9   13256       4  14846    0  1467490094    02/   
10  13256  421042  14847    0  1467530783    02/   
14  13252       1  14851    0  1467726894    01/   
83  13286     554  14927    0  1468423282    01/   

                                              comment  reply_to  
8   <iframe width="560" height="315" src="https://...         0  
9   that is a very heavy tail.  I got a similar re...         0  
10  I doubt it would have much effect on a large d...         0  
14  Hi, I haven't heard of someone doing this sort...         0  
83  Hi Sylvain,\r\n\r\nI don't know the answer to ...         0

In [269]:
top_aid_zero = q_resp_top_thread['aid'] == 0
q_resp_top_thread[top_aid_zero].shape, q_resp_top_thread[~top_aid_zero].shape
# ((2138, 8), (0, 8))

((2138, 8), (0, 8))

In [279]:
questions[questions['nid'] == 13256]

nid  uid  qid        date  user_cmnts  thread_cmnts  likes  \
1  13256    4    2  1467418233           2             3      0   

                                               title  
1  Question: Does the Levitation Light dive for y...

In [314]:
# top_aid_zero = q_resp_top_thread['aid'] == 0
# top_aid_zero = q_resp_top_thread['nid'] == 18937
# top_aid_zero = q_resp_top_thread['nid'] == 13256

q_resp_top_thread[top_aid_zero].shape[0]

2138

In [364]:
sub_aid_zero = q_resp_sub_threads['aid'] == 0
# q_resp_sub_threads[sub_aid_zero].sort_values('nid')

# q_resp_sub_threads[sub_aid_zero].shape, q_resp_sub_threads[~sub_aid_zero].shape
# ((13, 8), (799, 8))
# sub_aid_zero = q_resp_sub_threads['nid'] == 18937
# sub_aid_zero = q_resp_sub_threads['nid'] == 13256
# sub_aid_zero = q_resp_sub_threads['nid'] == 18930

q_resp_sub_threads[sub_aid_zero]['nid'].unique()#.sort_values('date')

array([18765, 18795, 18527, 18841, 18930, 18937, 18958])

In [369]:
c_not_in_a[c_not_in_a['nid'] == 18958]

nid  uid    cid  aid        date thread  \
7766  18958    1  23674    0  1554412505    01/   

                                                comment  reply_to  
7766   Hi @read_holman - there's some information (n...         0

In [362]:
comments[comments['cid'] == 23674]

nid  uid    cid  aid        date thread  \
7766  18958    1  23674    0  1554412505    01/   

                                                comment  reply_to  
7766   Hi @read_holman - there's some information (n...         0

In [363]:
answers[answers['nid'] == 18958]

Empty DataFrame
Columns: [nid, uid, aid, created_at, likes, content, accepted]
Index: []

In [308]:
answers[answers['nid'] == 18930]

Empty DataFrame
Columns: [nid, uid, aid, created_at, likes, content, accepted]
Index: []

In [307]:
questions[questions['nid'] == 18930]

nid     uid  qid        date  user_cmnts  thread_cmnts  likes  \
816  18930  579821  817  1554140953           2             5      1   

                                                 title  
816  Have you tried any good DIY microscope dyes or...

In [300]:
print notes[notes['nid'] == 18930]

nid     uid        date  comments  likes  \
2088  18930  579821  1554140953         5      1   

                                                  title  
2088  Have you tried any good DIY microscope dyes or...

In [313]:
answers.shape[0]

1080

In [ ]:
 	Edit Edit	Copy Copy	Delete Delete	note
	Edit Edit	Copy Copy	Delete Delete	page
	Edit Edit	Copy Copy	Delete Delete	event
	Edit Edit	Copy Copy	Delete Delete	feature
	Edit Edit	Copy Copy	Delete Delete	map
	Edit Edit	Copy Copy	Delete Delete	place
	Edit Edit	Copy Copy	Delete Delete	plots_aggregator
	Edit Edit	Copy Copy	Delete Delete	tool
	Edit Edit	Copy Copy	Delete Delete	to_do